src/sa.py

```diff
@@ -1,15 +1,194 @@
-import argparse
-
-
-def main():
-    argparser = argparse.ArgumentParser(
-        description="Exact matching using a suffix array")
-    argparser.add_argument("genome", type=argparse.FileType('r'))
-    argparser.add_argument("reads", type=argparse.FileType('r'))
-    args = argparser.parse_args()
-    print(f"Find every reads in {args.reads.name} " +
-          f"in genome {args.genome.name}")
-
-
-if __name__ == '__main__':
-    main()
+#####################
+# Libraries:
+
+import sys
+import random
+import math
+
+#####################
+# Functions:
+
+from cigar import edits_to_cigar
+from align import get_edits
+    
+# Simulate string:
+def simulate_string(m):
+    """Simulate a DNA sequence of length m.
+    
+    Example:
+    simulate_string(20)
+    
+    """
+    DNA = 'ACGT'
+    nucleotides = [random.choice(DNA) for _ in range(m)]
+    return ''.join(nucleotides)
+    
+
+# Naive approach
+def get_suffix_array(string):
+    string+='$'
+    SA = sorted(range(len(string)), key=lambda i: string[i:])
+    return SA

```

Comment:
Ok, yes. The slicing makes it slower than it needs to be, theoretically, but it moves the comparison to C, which more than compensates for it.